src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -96,15 +111,17 @@ private FolderComposite FindInputFiles(string path, List<string> filesToExclude,
                 FullPath = Path.GetFullPath(path),
                 RelativePath = parentFolder is null ? lastPathComponent : Path.Combine(parentFolder, lastPathComponent)
             };
-            foreach (var folder in _fileSystem.Directory.EnumerateDirectories(folderComposite.FullPath).Where(x => !_foldersToExclude.Contains(Path.GetFileName(x))))
-            {
-                folderComposite.Add(FindInputFiles(folder, filesToExclude, folderComposite.RelativePath));
-            }
-            foreach (var file in _fileSystem.Directory.GetFiles(folderComposite.FullPath, "*.cs", SearchOption.TopDirectoryOnly))
+
+            // If directory actually exists, search it for files
+            if (_fileSystem.Directory.Exists(folderComposite.FullPath))
             {
-                // Roslyn cannot compile xaml.cs files generated by xamarin. 
-                // Since the files are generated they should not be mutated anyway, so skip these files.
-                if (!file.EndsWith(".xaml.cs"))
+                foreach (var folder in _fileSystem.Directory.EnumerateDirectories(folderComposite.FullPath).Where(x => !_foldersToExclude.Contains(Path.GetFileName(x))))
```

Comment:
Can we add the "virtual" paths here to the real paths? Just append the linked files dir if it also starts with the same string. We could reuse all other logic in this method for both real and linked files.